prov/sm2/src/sm2_msg.c

```diff
@@ -38,6 +38,15 @@
 #include "ofi_iov.h"
 #include "sm2.h"
 
+static inline int sm2_select_proto(uint64_t total_len)
+{
+	if (total_len <= SM2_INJECT_SIZE) {
+		return sm2_proto_inject;
+	}
+
+	return sm2_proto_cma;
```

Comment:
are you planning on having some sort of similar cma check for sm2/will that come later?

Reply:
Yes... I want to have a 1 sided CMA check.

Reply:
Nit: it could be reformatted to

```
return total_len <= SM2_INJECT_SIZE ? sm2_proto_inject : sm2_proto_cma; 
```

I guess we are not doing that now as there are more protocols to come in?

Reply:
more protocols to come.